### DISCUSSION TITLE
Sensor amnesia | events not firing.

### DISCUSSION THREAD

**Discussion by myradon** [Koenkk/zigbee2mqtt#8162] [General]
Hey,

Recently I've update all my docker containers to latest stable: HA 2021.7.4, Z2M 1.20.0 (and more but not applicable I assume).

Aqara sensors motion, contact and wall switches seems to behave really odd. sometimes I have to trigger wall switch twice. Door contact events don't show up in Mosquitto. So have to open en close door twice than it behaves like expected But idle for X amount of time seems recreate issue again. Eg. 'contact_closet_hall' (in log)

Motion sensor illuminance doesn't update as much as before. So when  I want to trigger lights on depending on illuminance they stay off (done in Node-RED). Checking value in HA seems value hasn't been updated for several of hours. 

I have set availability_timeout: 60. 

I don't know whats going on and how to fix it.

[z2m_2021-07-28.11-57-13.log](https://github.com/Koenkk/zigbee2mqtt/files/6892364/z2m_2021-07-28.11-57-13.log)

**Comment by netweaver1970** [comment]
this (the need to double tap a switch or a longer delay after motion trigger) is a common issue lots of people (me included) have with a lot of wall switches/motion detectors. And there has never been a solution for, as far as I know. 
On the other hand, the illuminace update only happens, as far as I recall, only when there's a motion detected and -kind of hourly-. not contstantly. So if you need that, better go for true luminance sensors, not the combined motion/illuminance Aqara ones. 
This is one of my graphs, with no movement the whole day, so fully 'autonomous' reporting: 
![image](https://user-images.githubusercontent.com/18028618/127388613-bd090366-b740-4118-9688-e6070e16b181.png)

**Reply by myradon** [reply]
The problem with sending/triggering  events was non-existing before updates. Now nothing works like it should. Only non-Z2M sensors (ESPHome Motion) work fine through MQTT.

**Comment by myradon** [comment]
I think it has somehow to do with forgetting State after an X period of time. Because behaviour seems to me pointing that way.;After ... timespan event single_click twice seems to trigger normal automation, then with regular single_click event. I think similar with Aqara door sensors but not 100% accurate. Xiaomi Aqara Motion sensor I haven't traced a pattern yet.
